File: 2.1 Algorithmic Toolbox/Week 3 Greedy Algorithms/02_greedy_algorithms_starter_files/different_summands/different_summands.cpp
#include <iostream>
#include <vector>

using std::vector;

vector<int> optimal_summands(int n) {
  vector<int> summands;
  summands[0] = 1;
  int Test = 2;
  int Total = 3;
  while (Total <=  n){
	  summands.push_back(Test);
	  Test += 1;
	  Total += Test;
  }
  summands[-1] += n - (Total - Test);
  return summands;
}

int main() {
  int n;
  std::cin >> n;
  vector<int> summands = optimal_summands(n);
  std::cout << summands.size() << '\n';
  for (size_t i = 0; i < summands.size(); ++i) {
    std::cout << summands[i] << ' ';
  }
}

File: 2.1 Algorithmic Toolbox/Week 3 Greedy Algorithms/02_greedy_algorithms_starter_files/different_summands/different_summands.py
# Uses python3
import sys

def optimal_summands(n):
    summands = [1]
    Test = 2
    Total = 3
    while (Total <= n):
        summands.append(Test)
        Test += 1
        Total += Test
    summands[-1] += n - (Total - Test)
    return summands

if __name__ == '__main__':
    input = sys.stdin.read()
    n = int(input)
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print(x, end=' ')
